projects/plugins/jetpack/class.json-api-endpoints.php

```diff
@@ -1673,49 +1673,50 @@ public function get_media_item_v1_1( $media_id, $media_item = null, $file = null
-				? $attachment_metadata['filesize']
-				: 0;
+			$attachment_metadata = isset( $media_item->ID ) ? wp_get_attachment_metadata( $media_item->ID ) : array();
+			$filesize            = ! empty( $attachment_metadata['filesize'] ) ? $attachment_metadata['filesize'] : 0;
 		} else {
 			// For VideoPress videos, $attachment_file is the video URL.
-			$filesize = file_exists( $attachment_file )
-				? filesize( $attachment_file )
-				: 0;
+			$filesize = ( $attachment_file && file_exists( $attachment_file ) )
+			? filesize( $attachment_file )
+			: 0;
 		}
 
 		$response = array(
-			'ID'          => $media_item->ID,
-			'URL'         => wp_get_attachment_url( $media_item->ID ),
-			'guid'        => $media_item->guid,
-			'date'        => (string) $this->format_date( $media_item->post_date_gmt, $media_item->post_date ),
-			'post_ID'     => $media_item->post_parent,
-			'author_ID'   => (int) $media_item->post_author,
+			'ID'          => isset( $media_item->ID ) ? $media_item->ID : null,
+			'URL'         => isset( $media_item->ID ) ? wp_get_attachment_url( $media_item->ID ) : null,
+			'guid'        => isset( $media_item->guid ) ? $media_item->guid : null,
+			'date'        => ( isset( $media_item->post_date_gmt ) && isset( $media_item->post_date ) ) ?
+			(string) $this->format_date( $media_item->post_date_gmt, $media_item->post_date ) : null,
+			'post_ID'     => isset( $media_item->post_parent ) ? $media_item->post_parent : null,
+			'author_ID'   => isset( $media_item->post_author ) ? (int) $media_item->post_author : null,
 			'file'        => $file,
-			'mime_type'   => $media_item->post_mime_type,
+			'mime_type'   => isset( $media_item->post_mime_type ) ? $media_item->post_mime_type : null,
 			'extension'   => $ext,
-			'title'       => $media_item->post_title,
-			'caption'     => $media_item->post_excerpt,
-			'description' => $media_item->post_content,
-			'alt'         => get_post_meta( $media_item->ID, '_wp_attachment_image_alt', true ),
-			'icon'        => wp_mime_type_icon( $media_item->ID ),
+			'title'       => isset( $media_item->post_title ) ? $media_item->post_title : '',
+			'caption'     => isset( $media_item->post_excerpt ) ? $media_item->post_excerpt : '',
+			'description' => isset( $media_item->post_content ) ? $media_item->post_content : '',
+			'alt'         => isset( $media_item->ID ) ? get_post_meta( $media_item->ID, '_wp_attachment_image_alt', true ) : '',
+			'icon'        => isset( $media_item->ID ) ? wp_mime_type_icon( $media_item->ID ) : null,
 			'size'        => size_format( (int) $filesize, 2 ),
 			'thumbnails'  => array(),
 		);
 
-		if ( in_array( $ext, array( 'jpg', 'jpeg', 'png', 'gif', 'webp' ), true ) ) {
+		if ( in_array( $ext, array( 'jpg', 'jpeg', 'png', 'gif', 'webp' ), true ) && isset( $media_item->ID ) ) {
 			$metadata = wp_get_attachment_metadata( $media_item->ID );
-			if ( isset( $metadata['height'], $metadata['width'] ) ) {
+			if ( isset( $metadata['height'] ) ) {
```

Comment:
No warnings were being generated for this, other than a codesniffer message suggesting`isset` should only check one parameter at a time, hence the change. Same further down as well (around line 1774).